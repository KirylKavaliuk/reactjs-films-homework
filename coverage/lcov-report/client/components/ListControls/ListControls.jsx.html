<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/ListControls/ListControls.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/ListControls</a> ListControls.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
&nbsp;
import classNames from 'classnames';
&nbsp;
import { getSection } from 'utils/url';
&nbsp;
import Select from 'components/Select/Select';
import Icon from 'components/Icon/Icon';
import Link from 'components/Link/Link';
&nbsp;
import styles from './ListControls.scss';
&nbsp;
export default class ListControls extends Component {
  static setActiveStyleForLinks = (sectionNumber) =&gt; {
    const section = getSection();
    const sections = ['trading', 'top-rated', 'coming-soon'];
    const index = sections.findIndex(_section =&gt; section.indexOf(_section) !== -1);
    const defaultSection = section === '/' &amp;&amp; sectionNumber === 0;
&nbsp;
    return classNames(
      styles.section,
      { [styles.activeLink]: index === sectionNumber || defaultSection },
    );
  }
&nbsp;
  static setTypeViewClasses = gridView =&gt; (
    classNames(styles.typeView, { [styles.activeTypeView]: gridView })
  )
&nbsp;
  render() {
    const { match, genres, gridView } = this.props;
&nbsp;
    return (
      &lt;menu className={ styles.menu }&gt;
        &lt;div className={ styles.listControls }&gt;
          &lt;ul className={ styles.sections }&gt;
            &lt;Link to='/trading' clearParams={['query', 'movie']}&gt;
              &lt;li className={ ListControls.setActiveStyleForLinks(0) }&gt;Trading&lt;/li&gt;
            &lt;/Link&gt;
            &lt;Link to='/top-rated' clearParams={['query', 'movie']}&gt;
              &lt;li className={ ListControls.setActiveStyleForLinks(1) }&gt;Top Rated&lt;/li&gt;
            &lt;/Link&gt;
            &lt;Link to='/coming-soon' clearParams={['query', 'movie']}&gt;
              &lt;li className={ ListControls.setActiveStyleForLinks(2) }&gt;Coming soon&lt;/li&gt;
            &lt;/Link&gt;
          &lt;/ul&gt;
          &lt;Select match={ match } genres={ genres }/&gt;
        &lt;/div&gt;
        &lt;div className={ styles.typesView }&gt;
          &lt;Link params={{ view: 'grid' }}&gt;
            &lt;div className={ ListControls.setTypeViewClasses(gridView) }&gt;
              &lt;Icon name='grid-view'/&gt;
            &lt;/div&gt;
          &lt;/Link&gt;
          &lt;Link params={{ view: 'list' }}&gt;
            &lt;div className={ ListControls.setTypeViewClasses(!gridView) }&gt;
              &lt;Icon name='list-view'/&gt;
            &lt;/div&gt;
          &lt;/Link&gt;
        &lt;/div&gt;
      &lt;/menu&gt;
    );
  }
}
&nbsp;
ListControls.defaultProps = {
  match: {},
  genres: [],
  gridView: true,
};
&nbsp;
ListControls.propTypes = {
  match: PropTypes.object,
  genres: PropTypes.array,
  gridView: PropTypes.bool,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
