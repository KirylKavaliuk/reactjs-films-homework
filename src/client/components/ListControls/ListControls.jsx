import React, { Component } from 'react';
import Link from 'components/Link/Link';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import { getSection } from 'utils/url';

import Select from 'components/Select/Select';
import Icon from 'components/Icon/Icon';

import styles from './ListControls.scss';

export default class ListControls extends Component {
  setActiveStyleForLinks(sectionNumber) {
    const section = getSection();
    const sections = ['trading', 'top-rated', 'coming-soon'];
    const index = sections.findIndex(_section => section.indexOf(_section) !== -1);
    const defaultSection = section === '/' && sectionNumber === 0;

    const that = this;

    return classNames(
      styles.section,
      { [styles.activeLink]: index === sectionNumber || defaultSection },
    );
  }

  render() {
    return (
      <menu className={ styles.menu }>
        <div className={ styles.listControls }>
          <ul className={ styles.sections }>
            <Link to='/trading' clearParams={['query', 'movie']}>
              <li className={ this.setActiveStyleForLinks(0) }>Trading</li>
            </Link>
            <Link to='/top-rated' clearParams={['query', 'movie']}>
              <li className={ this.setActiveStyleForLinks(1) }>Top Rated</li>
            </Link>
            <Link to='/coming-soon' clearParams={['query', 'movie']}>
              <li className={ this.setActiveStyleForLinks(2) }>Coming soon</li>
            </Link>
          </ul>
          <Select
            match={ this.props.match }
            genres={ this.props.genres }
          />
        </div>
        <div className={ styles.typesView }>
          <Link params={{ view: 'grid' }}>
            <div
              className={ classNames(
                styles.typeView,
                { [styles.activeTypeView]: this.props.gridView },
              ) }
            >
              <Icon name='grid-view'/>
            </div>
          </Link>
          <Link params={{ view: 'list' }}>
            <div
              className={ classNames(
                styles.typeView,
                { [styles.activeTypeView]: !this.props.gridView },
              ) }
            >
              <Icon name='list-view'/>
            </div>
          </Link>
        </div>
      </menu>
    );
  }
}

ListControls.defaultProps = {
  match: {},
  genres: [],
  gridView: true,
};

ListControls.propTypes = {
  match: PropTypes.object,
  genres: PropTypes.array,
  gridView: PropTypes.bool,
};
