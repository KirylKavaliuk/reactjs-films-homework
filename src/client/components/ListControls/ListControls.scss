@import 'styles/palette';

.menu {
  display: flex;
  border-bottom: 1px $LIGHT_GREY solid;
  justify-content: space-between;
}

.list-controls {
  position: relative;
  display: flex;
  min-width: 500px;
}

.sections {
  display: flex;
  color: $LIGHT_BLUE;
  align-items: center;
  list-style: none;

  a {
    text-decoration: none;
  }
}

.section {
  box-sizing: border-box;
  padding: 8px 10px;
  margin: 0 5px;

  &:first-child {
    padding-left: 0;
  }
}

.active-link {
  box-sizing: border-box;
  border-top: 2px $LIGHTER_GREY solid;
  border-bottom: 2px $LIGHT_BLUE solid;
}

.types-view {
  display: flex;
  height: 25px;
  justify-content: space-between;
}

.type-view {
  width: 25px;
  height: 25px;
  margin-left: 5px;
  transition: all 0.45s ease;
  fill: $LIGHT_GREY;

  &:hover {
    cursor: pointer;
  }
}

.active-type-view {
  fill: $LIGHT_BLUE;
}

@media only screen and (max-width: 650px) {
  .menu {
    flex-wrap: wrap-reverse;
  }

  .types-view {
    width: 100%;
    justify-content: flex-end;
  }
}

@media only screen and (max-width: 560px) {
  .movies-list {
    padding: 20px;
  }

  .list-controls {
    min-width: 100%;
    flex-wrap: wrap;
  }

  .menu-links {
    width: 100%;
    justify-content: space-between;
  }

  .menu-link {
    padding: 5px 10px;

    &:first-child {
      padding: 5px 10px;
    }
  }
}

@media only screen and (max-width: 430px) {
  .movies-list {
    padding: 10px;
  }

  .menu-links {
    flex-wrap: wrap;
  }

  .menu-link {
    width: 100%;
    margin: 5px 0;
    padding: 5px 10px;
  }
}
