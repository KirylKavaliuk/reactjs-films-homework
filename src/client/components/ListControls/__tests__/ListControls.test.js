import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import ListControls from '../ListControls';


it('renders list controls correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <ListControls
        genres={ [
          { id: 1, name: 'genre1' },
          { id: 2, name: 'genre2' },
          { id: 3, name: 'genre3' },
        ] }
        match={{
          params: { genreId: 2 },
          isExact: true,
          path: '',
          url: '',
        }}
      />
    </MemoryRouter>,
  );

  const listControls = tree.toJSON();

  expect(listControls).toMatchSnapshot();
});
