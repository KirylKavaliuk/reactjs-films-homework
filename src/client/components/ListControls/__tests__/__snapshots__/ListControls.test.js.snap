// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list controls correctly 1`] = `
<menu
  className="ListControls__menu___3hmgJ"
>
  <div
    className="ListControls__list-controls___3Tclz"
  >
    <ul
      className="ListControls__sections___38v7z"
    >
      <a
        className="Link__link___-IU0S"
        href="/trading"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="ListControls__section___1rlYr ListControls__active-link___L0qDL"
        >
          Trading
        </li>
      </a>
      <a
        className="Link__link___-IU0S"
        href="/top-rated"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="ListControls__section___1rlYr"
        >
          Top Rated
        </li>
      </a>
      <a
        className="Link__link___-IU0S"
        href="/coming-soon"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="ListControls__section___1rlYr"
        >
          Coming soon
        </li>
      </a>
    </ul>
    <div
      className="Select__select___1ZlRu"
      onMouseLeave={[Function]}
    >
      <div
        className="Select__main___3EP3d"
        onClick={[Function]}
      >
        <span
          className="Select__value___2xWIB"
        >
          Genre
        </span>
        <svg
          className="Select__arrow___kPPA1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </div>
      <div
        className="Select__list___2Ii_p"
      >
        <a
          className="Link__link___-IU0S Select__item____FX41"
          href="/genre/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre1
        </a>
        <a
          className="Link__link___-IU0S Select__item____FX41"
          href="/genre/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre2
        </a>
        <a
          className="Link__link___-IU0S Select__item____FX41"
          href="/genre/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre3
        </a>
      </div>
    </div>
  </div>
  <div
    className="ListControls__types-view___tY9PL"
  >
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="ListControls__type-view___26jLs ListControls__active-type-view___3Owd9"
      >
        <svg
          className=""
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM8 20H4v-4h4v4zm0-6H4v-4h4v4zm0-6H4V4h4v4zm6 12h-4v-4h4v4zm0-6h-4v-4h4v4zm0-6h-4V4h4v4zm6 12h-4v-4h4v4zm0-6h-4v-4h4v4zm0-6h-4V4h4v4z"
          />
        </svg>
      </div>
    </a>
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="ListControls__type-view___26jLs"
      >
        <svg
          className=""
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 13H3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h17c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1zm0-10H3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h17c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1z"
          />
        </svg>
      </div>
    </a>
  </div>
</menu>
`;
